contracts/core/EntryPointSimulations.sol

```diff
@@ -112,16 +116,51 @@ contract EntryPointSimulations is EntryPoint, IEntryPointSimulations {
         if (target != address(0)) {
             (targetSuccess, targetResult) = target.call(targetCallData);
         }
-        return ExecutionResult(
-            opInfo.preOpGas,
-            paid,
+        uint userOpTotalValidationGasUsed = opInfo.preOpGas - op.preVerificationGas;
+        //userOpActualGasUsed, userOpSuccess are saved by _emitUserOperationEvent, below..
```

Comment:
Maybe `userOpActualGasUsed` and `userOpSuccess` should not be different from `userOpPostOpGas`?
Make a call to an empty callback in the deployed code. Instead of creating two distinct `_emitUserOperationEvent` functions, for simulation and execution?

Reply:
the _emit method will stay, since it so happens that emitting from a method is slightly more gas-efficient than inlining it...
Do you think we should pass those parameters twice, once to the _emit method and another to a debug-only method?